src/services/expensesServices.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+let expenses = [];
+
+const getAllExpenses = (userId, categories, from, to) => {
+  let preparedExpenses = expenses;
+
+  if (userId) {
+    preparedExpenses
+      = preparedExpenses.filter((expense) => expense.userId === +userId);
+  }
+
+  if (categories) {
+    preparedExpenses = preparedExpenses.filter((expense) =>
+      categories.includes(expense.category)
+    );
+  }
+
+  if (from && to) {
+    preparedExpenses = preparedExpenses.filter(
+      (expense) =>
+        new Date(expense.spentAt) >= new Date(from)
+        && new Date(expense.spentAt) <= new Date(to)
+    );
+  }
+
+  return preparedExpenses;
+};
+
+const getExpenseById = (id) => {
+  return expenses.find(expense => expense.id === +id) || null;
```

Comment:
<img width="781" alt="image" src="https://github.com/mate-academy/node_accounting-app-v2/assets/54860641/5573b8a8-06ca-487f-bb41-b1d60c2e9e99">

```suggestion
  return users.find(user => user.id === +id);
```